src/frontend/src/lib/utils/token.utils.ts

```diff
@@ -131,3 +139,108 @@ export const mapTokenUi = ({
 		$exchanges
 	})
 });
+
+/**
+ * Type guard to check if a token is of type RequiredTokenWithLinkedData.
+ * This checks whether the token has a twinTokenSymbol field and ensures that it is a string.
+ *
+ * @param token - The token object to be checked.
+ * @returns A boolean indicating whether the token is of type RequiredTokenWithLinkedData.
+ */
+export const isRequiredTokenWithLinkedData = (token: Token): token is RequiredTokenWithLinkedData =>
+	'twinTokenSymbol' in token && typeof token.twinTokenSymbol === 'string';
+
+/**
+ * Type guard to check if an object is of type TokenGroupUi.
+ *
+ * @param obj - The object to check.
+ * @returns A boolean indicating whether the object is a TokenGroupUi.
+ */
+export const isTokenGroupUi = (obj: unknown): obj is TokenGroupUi =>
+	typeof obj === 'object' &&
+	obj !== null &&
+	'header' in obj &&
+	typeof (obj as TokenGroupUi).header === 'object' &&
+	'tokens' in obj;
+
+/**
+ * Factory function to create a TokenGroupUi based on the provided tokens and network details.
+ * This function creates a group header and adds both the native token and the twin token to the group's tokens array.
+ *
+ * @param nativeToken - The native token used for the group, typically the original token or the one from the selected network.
+ * @param twinToken - The twin token to be grouped with the native token, usually representing the same asset on a different network.
+ *
+ * @returns A TokenGroupUi object that includes a header with network and symbol information and contains both the native and twin tokens.
+ */
+const createTokenGroup = ({
+	nativeToken,
+	twinToken
+}: {
+	nativeToken: TokenUi;
+	twinToken: TokenUi;
+}): TokenGroupUi => ({
+	header: {
+		name: nativeToken.network.name,
+		symbol: `${nativeToken.symbol}, ${twinToken.symbol}`,
+		decimals: nativeToken.decimals,
+		icon: nativeToken.icon ?? '/images/default_token_icon.svg'
```

Comment:
here it would be better to use the component that is a placeholder for logo, I think it is `Logo` or `IconRandom`. Or maybe it is not even necessary to add a fallback, since it is handled in the card (we usually don't add an icon fallback)

Reply:
5664d75a